### ISSUE TITLE
Hard to debug TypeError: arguments[i].apply is not a function

### ISSUE THREAD

**Issue by emilio-martinez** [webpack/webpack#6003]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

This is a means to reopening "Hard to debug TypeError: arguments[i].apply is not a function " (https://github.com/webpack/webpack/issues/2217). That issue was closed due to inactivity, but I encountered it again.

As a quick recap, I run webpack (in my case via CLI), and very soon after get:

```
PROJECT_PATH/node_modules/tapable/lib/Tapable.js:375
		arguments[i].apply(this);
               ^
TypeError: arguments[i].apply is not a function
    at Compiler.apply (PROJECT_PATH/node_modules/tapable/lib/Tapable.js:375:16)
    at webpack (PROJECT_PATH/node_modules/webpack/lib/webpack.js:33:19)
    at processOptions (PROJECT_PATH/node_modules/webpack/bin/webpack.js:335:15)
    at yargs.parse (PROJECT_PATH/node_modules/webpack/bin/webpack.js:393:2)
    at Object.Yargs.self.parse (PROJECT_PATH/node_modules/webpack/node_modules/yargs/yargs.js:533:18)
    at Object.<anonymous> (PROJECT_PATH/node_modules/webpack/bin/webpack.js:152:7)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3
```

**If the current behavior is a bug, please provide the steps to reproduce.**

The issue as described in https://github.com/webpack/webpack/issues/2217 is, in my opinion, valid, as it is in fact hard to debug. I described how to reproduce it above, but additionally, I can report what was causing it—to be clear, that's what fixed it in my issue, but because the error message is ambiguous, it could also occur for other reasons.

In my case, I accidentally had pasted a non-plugin object into the `plugins` array, which I ultimately figured it out via trial and error.

**What is the expected behavior?**

A more helpful error message. Since the error happens on `arguments[i].apply(this);` perhaps reporting that it happened when parsing the `plugins` and providing the index relative the `plugins` array that's causing the error?

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

```
node 6.11.1
webpack 3.8.1
```

**Comment by EugeneHlushko** [comment]
@emilio-martinez i hear your pain, fix in #6008 if passes pr

**Comment by ooflorent** [comment]
Fixed by #6008
